os/arch/arm/src/common/up_createstack.c

```diff
@@ -243,6 +252,18 @@ int up_create_stack(FAR struct tcb_s *tcb, size_t stack_size, uint8_t ttype)
 		heapinfo_exclude_stacksize(tcb->stack_alloc_ptr);
 #endif
 		board_led_on(LED_STACKCREATED);
+#ifdef CONFIG_MPU_STACKGUARD
+		/* The smallest size that can be programmed for an MPU region is 32 bytes */
+
+		tcb->stack_guard =	(uint32_t)tcb->stack_alloc_ptr + 0x40;
```

Comment:
It would be better to define a new alignment directive like STACK_ALIGN_UP to perform the proper 32 byte alignment.

Reply:
added new local alignment directive